### PR TITLE
Bug 1523403 - Test coverage for DS front-end

### PR DESCRIPTION
**Pull request by punamdahiya**

### CODE REVIEW

File: test/unit/content-src/components/DiscoveryStreamComponents/DSLinkMenu.test.jsx [hunk 1]
```diff
@@ -13,6 +13,28 @@ describe("<DSLinkMenu>", () => {
...
   });
 
+  it("on Menu Update", () => {
+    const dsLinkMenu = mountWithIntl(<DSLinkMenu />);
+    dsLinkMenu.setState({showContextMenu: true});
+
+    dsLinkMenu.instance().onMenuUpdate(false);
+    dsLinkMenu.update();
+
+    // Should check grand parent element is missing active, how?
```
Comment (punamdahiya) [inline test/unit/content-src/components/DiscoveryStreamComponents/DSLinkMenu.test.jsx]: @Mardak is there any way to get div (e.g.  .ds-card) containing DSLinkMenu here. Thought should ask you before moving test to check for "active" and "last-item" class in DSCard component

Comment (Mardak) [reply test/unit/content-src/components/DiscoveryStreamComponents/DSLinkMenu.test.jsx]: The behavior of DSLinkMenu modifying ancestors does look strange, but ignoring that, you can use `dsLinkMenu.getDOMNode().parentNode.classList` to check if "active" is in there (for the "onMenuShow" test). For here "onMenuUpdate", you can do the opposite by adding the class name after mounting then checking it's removed.

File: test/unit/content-src/components/DiscoveryStreamComponents/TopSites.test.jsx [hunk 2]
```diff
@@ -23,3 +27,20 @@ describe("Discovery Stream <TopSites>", () => {
...
+describe.only("DSCard DS Top Sites", () => {
+  let wrapper;
+  let DEFAULT_PROPS = {
+    header: "test",
+    isFixed: true,
+  };
+
+  beforeEach(() => {
+    const store = createStore(combineReducers(reducers), INITIAL_STATE);
+    wrapper = mountWithIntl(<Provider store={store}><TopSites {...DEFAULT_PROPS} /></Provider>);
```
Comment (punamdahiya) [inline test/unit/content-src/components/DiscoveryStreamComponents/TopSites.test.jsx]: @Mardak Discovery stream component has 75% test coverage calling out connect as uncovered block

Mounting Topsites above is giving error here https://github.com/mozilla/activity-stream/blob/master/content-src/components/TopSites/TopSite.jsx#L418
with this.props.TopSitesRows as undefined. Looks like its set  https://github.com/mozilla/activity-stream/blob/master/content-src/components/TopSites/TopSites.jsx#L172

Is there any way to set state.Prefs.values.topSitesRows from tests to override this error? Thanks

top-site-code-coverage

**Comment by punamdahiya** [comment]
@Mardak @ScottDowne PR increases test coverage for DS components to 90%. Please review. Thanks!

**Comment by punamdahiya** [comment]
@gvn if you have bandwidth, please review. Thanks!

File: test/unit/content-src/components/DiscoveryStreamComponents/TopSites.test.jsx (hunk 2 above)
Comment (Mardak) [reply test/unit/content-src/components/DiscoveryStreamComponents/TopSites.test.jsx]: Just to note out conversation from slack, we need to import the 2 differently exported components to test the connected version vs plain component.

Comment (punamdahiya) [reply test/unit/content-src/components/DiscoveryStreamComponents/TopSites.test.jsx]: @Mardak quick clarification on the feedback, do we want separate unit tests for connected (TopSites) and plain (_TopSites) component. My understanding mounting connected component loads and run tests on plain component, and showing test coverage 100% for TopSites component

**Comment by punamdahiya** [comment]
Synched up with @Mardak and TopSites testing of the connected version also tests the wrapped unconnected version and looks good.

**Pull request merged**